exercise/submission_models.py

```diff
@@ -154,6 +156,10 @@ def create_from_post(self, exercise, submitters, request):
         if 'lang' not in meta_data_dict:
             meta_data_dict['lang'] = get_language()
 
+        #create Hash for the new submission
+        new_hash = helpers.get_hash(submission_data_list, request.FILES)
```

Comment:
This computes the hash without the `__aplus__` key in the POST data. I think the JS code chapter.js includes those keys in its hash computation. This is one reason why the hashes become different in the frontend and backend.

The `get_hash()` function definition uses `*params`, but this line calls the function with a normal list without unpacking it.